### PR TITLE
chore: public client react example

### PR DESCRIPTION
**Pull request by NithinKuruba**

### CODE REVIEW

File: public-react/src/services/keycloak.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import Keycloak from 'keycloak-js';
+
+const _kc = new Keycloak({
+  url: 'https://dev.loginproxy.gov.bc.ca/auth',
```
Comment (junminahn) [inline public-react/src/services/keycloak.js]: can we make this an env as well?

File: public-react/src/services/keycloak.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  realm: process.env.REACT_APP_REALM,
+  clientId: process.env.REACT_APP_CLIENT_ID,
+});
+
+const loginOptions = {
+  redirectUri: `https://logon7.gov.bc.ca/clp-cgi/logoff.cgi?retnow=1&returl=${encodeURIComponent(
+    process.env.REACT_APP_URL,
```
Comment (junminahn) [inline public-react/src/services/keycloak.js]: can rename the env to indicate it is more specific to redirect uri? e.g. `REACT_SSO_REDIRECT_URI`

File: public-react/src/services/keycloak.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  )}`,
+  idpHint: '',
+};
+
+export const initializeKeycloak = async () => {
+  try {
+    const auth = await _kc.init({
+      pkceMethod: 'S256',
+      checkLoginIframe: false,
```
Comment (junminahn) [inline public-react/src/services/keycloak.js:25]: as we disable `checkLoginIframe` option, can we have a logic to refresh the token before expires?
something like: https://github.com/bcgov/keycloak-example-apps/blob/dev/public-spa/pages/_app.tsx#L43

File: public-react/src/services/keycloak.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import Keycloak from 'keycloak-js';
+
+const _kc = new Keycloak({
+  url: 'https://dev.loginproxy.gov.bc.ca/auth',
+  realm: process.env.REACT_APP_REALM,
```
Comment (junminahn) [inline public-react/src/services/keycloak.js]: can we rename the env `REACT_APP_REALM` to `REACT_SSO_REALM` to make it obvious that the env is being used for login (SSO) purpose?

File: public-react/src/services/keycloak.js (hunk 1 above)
Comment (junminahn) [inline public-react/src/services/keycloak.js]: can we make this also an env? e.g. `REACT_SSO_URL`

File: public-react/src/services/keycloak.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  clientId: process.env.REACT_APP_CLIENT_ID,
+});
+
+const loginOptions = {
+  redirectUri: `https://logon7.gov.bc.ca/clp-cgi/logoff.cgi?retnow=1&returl=${encodeURIComponent(
```
Comment (junminahn) [inline public-react/src/services/keycloak.js:11]: can we add a comment on what this extra URL `https://logon7.gov.bc.ca` does in the codebase for them?

File: public-react/src/services/keycloak.js (hunk 1 above)
Comment (NithinKuruba) [reply public-react/src/services/keycloak.js]: done!

File: public-react/src/services/keycloak.js (hunk 2 above)
Comment (NithinKuruba) [reply public-react/src/services/keycloak.js]: done!

File: public-react/src/services/keycloak.js (hunk 3 above)
Comment (NithinKuruba) [reply public-react/src/services/keycloak.js:25]: done!

File: public-react/src/services/keycloak.js (hunk 4 above)
Comment (NithinKuruba) [reply public-react/src/services/keycloak.js]: @junminahn react env vars have to start with `REACT_APP` so I can rename it to `REACT_APP_SSO_REALM`

File: public-react/src/services/keycloak.js (hunk 5 above)
Comment (NithinKuruba) [reply public-react/src/services/keycloak.js:11]: done!

File: public-react/src/services/keycloak.js (hunk 4 above)
Comment (junminahn) [reply public-react/src/services/keycloak.js]: @NithinKuruba right; I think I meant that `REACT_APP_SSO_REALM`

**Review by junminahn — APPROVED** [review]

**Pull request merged**